src/test/java/org/apache/sling/feature/cpconverter/accesscontrol/AclManagerTest.java

```diff
@@ -120,6 +155,37 @@ public void makeSureAclsAreCreatedOnlyoutsideSytemUsersPaths() throws Exception
         assertFalse(operations.isEmpty());
     }
 
+    @Test
+    public void testAddACLforUnknownUser() throws RepoInitParsingException {
+        aclManager.addSystemUser(new SystemUser("acs-commons-package-replication-status-event-service", new RepoPath("/home/users/system/foo"), new RepoPath("/home/users/system")));
```

Comment:
wouldn't it be easier to omit that an only call 'addAcl' for an unknown principal? the repo-init statement should then be empty (or nearly empty)

Reply:
good idea - in that case the extension should be null because it doesn't have anything. I'll change that.